public/app/features/alerting/unified/components/rule-editor/alert-rule-form/AlertRuleForm.tsx

```diff
@@ -401,6 +404,30 @@ function formValuesFromPrefill(rule: Partial<RuleFormValues>): RuleFormValues {
   });
 }
 
+function setInstantOrRange(values: RuleFormValues): RuleFormValues {
+  return {
+    ...values,
+    queries: values.queries?.map((query) => {
+      if (!isExpressionQuery(query.model)) {
+        // data query
+        const defaultToInstant =
+          query.model.datasource?.type === 'loki' || query.model.datasource?.type === 'prometheus';
```

Comment:
Can we use `DataSourceType.Loki` and `DataSourceType.Prometheus`?

Reply:
💯 I was not aware of this enum